packages/gatsby-transformer-json/src/gatsby-node.js

```diff
@@ -10,6 +10,8 @@ async function onCreateNode({ node, actions, loadNodeContent, createNodeId }, pl
       return pluginOptions.typeName
     } else if (isArray) {
       return _.upperFirst(_.camelCase(`${node.name} Json`))
+    } else if (node.internal.type !== `File`) {
+      return `${node.internal.type} Json`
```

Comment:
i am not sure if this is the better option, as this dictates the graphql schema, e.g. the contentful sets `internal.type` to ``_.camelCase(`${node.internal.type} ${key} JSONNode`)`` so you would need to query by that name. do you have examples from contentful? what about `node.name`?

Reply:
Is `node.name` always available and suitable for GraphQL? I have no dog in the fight, just trying to get the incompatibility resolved so I can keep moving on a project.

Reply:
`node.name` is also `File` specific - but at least it won't crash if `name` is not set, but yeah, let's move this `else if` before `else if (isArray)`